src/decorators.py

```diff
@@ -0,0 +1,38 @@
+import logging
+from functools import wraps
+
+import pandas as pd
+
+from logging_config import setup_logging
+
+setup_logging()
+logger = logging.getLogger("my_log")
+
+
+def decorator_record_file(file_name):
+    """
+    Декоратор, который записывает результат выполнения функции в JSON файл. Принимает на вход имя файла.
+    """
+
+    def wrapper(func):
```

Comment:
Проверку на тайпхинты делать в декоратор немного не в стиле pyhon - это должна быть ответственность разработчика, так что лучше через тайпхинты:
```suggestion
    def wrapper(func: Callable[..., pd.DataFrame]):
```